src/server.ts

```diff
@@ -142,17 +143,20 @@ export class Server {
    *
    * Allows you to directly inspect the current state of a contract. This is a backup way to access your contract data which may not be available via events or simulateTransaction.
    *
-   * @param {string} contractId - The contract ID containing the data to load. Encoded as a hex string.
+   * @param {string} contractId - The contract ID containing the data to load. Encoded as Stellar Contract Address e.g. `CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5` or a hex string for backwards compatibility, but will likely be deprecated in the future.
    * @param {xdr.ScVal} key - The key of the contract data to load.
    * @returns {Promise<SorobanRpc.LedgerEntryResult>} Returns a promise to the {@link SorobanRpc.LedgerEntryResult} object with the current value.
    */
   public async getContractData(
     contractId: string,
     key: xdr.ScVal,
   ): Promise<SorobanRpc.LedgerEntryResult> {
+    const buf = contractId.startsWith("C")
+      ? Address.fromString(contractId).toBuffer()
+      : Buffer.from(contractId, "hex");
```

Comment:
It could also be a hex string starting with 'C'...

Reply:
Nice catch, you’d need a length check? Hacky but imo we do this for one release then drop hex support?

Reply:
Or we could add https://github.com/stellar/js-stellar-base/pull/614, and use `new Contract(contractId).address().toBuffer()`, which would handle the appropriate parsing.

Reply:
This `Contract.address()` helper was just released in `stellar-base@v9.0.0-soroban.2`, so if we update the `package.json`, we can use that.